tests/Dispatching/DispatchingSetTest.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace EquipTests\Dispatching;
+
+use Auryn\Injector;
+use Equip\Adr\DomainInterface;
+use Equip\Directory;
+use Equip\Exception\DispatchingException;
+use Equip\Dispatching\DispatchingSet;
+use PHPUnit_Framework_TestCase as TestCase;
+
+class DispatchingSetTest extends TestCase
+{
+    public function testWithInvalidEntries()
+    {
+        $this->setExpectedExceptionRegExp(
+            DispatchingException::class,
+            '/Dispatch .* is not invokable/i'
+        );
+
+        new DispatchingSet(['foo']);
+    }
+
+    public function testWithValidEntries()
+    {
+        $dispatchers = [
+            function () {
+            }
+        ];
+        $dispatching = new DispatchingSet($dispatchers);
+        $this->assertSame($dispatchers, $dispatching->toArray());
+    }
+
+    public function testPrepareDirectory()
```

Comment:
I think the point of this test should be to ensure that `DispatchingSet->__invoke()` is capable of returning an instance of `Directory` that is not the same as the one that was originally passed to it (i.e. because a dispatcher returns a modified copy of it).